### PR TITLE
#VGGGie9k Setup Redux

### PR DESCRIPTION
**Pull request by Ally4**
#### What does this PR do?

- This PR is about setting up of redux and a simple number counter on the landing page

#### Description of Task to be completed?

- Installation of the redux package

- Creation of the simple number counter

#### How should this be manually tested?

- By cloning this repository

- By running `npm install` for the packages

- By running `npm start` to start the server

- By clicking `Increment` or `Decrement` in the opened browser

#### Any background context you want to provide?

- N/A

#### What are the relevant Trello story?

- `https://trello.com/c/VGGGie9k/26-setup-redux`

#### Questions:

### CODE REVIEW

**Review by Mugema123 — DISMISSED** [review]

**Review by Ntwali-Josue — DISMISSED** [review]

**Review by TuyizeeAnastase — DISMISSED** [review]

File: src/index.html [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Phantoms</title>
+</head>
+<body>
+    <div id="root"></div>
+</body>
+</html>
```
Comment (blaise82) [inline src/index.html]: why do you have this here>?

File: src/redux/store/index.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { createStore, compose, applyMiddleware, combineReducers } from "redux";
+import thunk from "redux-thunk";
+import counterReducer from "../reducers/counter";
+
+const rootReducer = combineReducers({
+  count: counterReducer,
+});
```
Comment (blaise82) [inline src/redux/store/index.js]: this should be in the index of reducers

File: src/redux/actions/actionTypes.js [hunk 3]
```diff
@@ -0,0 +1,2 @@
+export const INCREMENT = 'INCREMENT';
+export const DECREMENT = 'DECREMENT';
```
Comment (blaise82) [inline src/redux/actions/actionTypes.js]: types should be in the types folder

File: src/components/App.js [hunk 4]
```diff
@@ -1,13 +1,32 @@
...
+
+const mapDispatchToProps = (dispatch) => {
+  return {
+    increment: () => dispatch(actions.increment()),
+    decrement: () => dispatch(actions.decrement()),
+  };
 };
 
-export default App;
+export default connect(mapStateToProps, mapDispatchToProps)(App);
```
Comment (blaise82) [inline src/components/App.js]: the app component should be in ./src

File: .babelrc [hunk 5]
```diff
@@ -6,4 +6,4 @@
   "plugins": [
       "@babel/plugin-proposal-class-properties"
```
Comment (blaise82) [inline .babelrc]: you also add @babel/plugin-transform-runtime

**Review by blaise82 — CHANGES_REQUESTED** [review]

File: src/components/App.js (hunk 4 above)
Comment (Ally4) [reply src/components/App.js]: done

File: .babelrc (hunk 5 above)
Comment (Ally4) [reply .babelrc]: done

File: src/redux/actions/actionTypes.js (hunk 3 above)
Comment (Ally4) [reply src/redux/actions/actionTypes.js]: done

File: src/redux/store/index.js (hunk 2 above)
Comment (Ally4) [reply src/redux/store/index.js]: done

File: src/index.html (hunk 1 above)
Comment (Ally4) [reply src/index.html]: it was a slight problem with rebasing, but I just solved it!.

**Review by Mugema123 — APPROVED** [review]

**Review by blaise82 — APPROVED** [review]

**Pull request merged**